### PR TITLE
fix(multiselect): items can be removed from `modelValue` [MA-2104]

### PR DESCRIPTION
**Pull request by adorack**
- Add code path to handle the case where items are removed from `modelValue` rather than added.
- Add test.

# Summary

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Tests coverage:** test coverage was added for new features and bug fixes
* [ ] **Docs:** includes a technically accurate README

### CODE REVIEW

**Comment by adamdehaven** [comment]
@adorack this will need to target the `alpha` branch

**Comment by adorack** [comment]
This is targeting `main` rather than `alpha` because it's a bug fix.  Let's say that you have a Multiselect that looks roughly like this:

```
<Multiselect :items="items" v-model="selectedItems" />
```

`selectedItems` is an array of string values.  If you add an item value to the array (or set an array with more item values, or whatever), the multiselect will update its DOM to reflect the change.  However, if you remove an item from the array, the multiselect will not update its DOM to reflect the change.  This is because the code path for detecting missing items wasn't present.  This PR adds that path.

File: src/components/KMultiselect/KMultiselect.vue [hunk 1]
```diff
@@ -641,6 +657,41 @@ const handleMultipleItemsSelect = (items: MultiselectItem[]) => {
   stageSelections()
 }
 
+const handleMultipleItemDeselect = (items: MultiselectItem[], restage = false) => {
```
Comment (portikM) [inline src/components/KMultiselect/KMultiselect.vue:660]:
```suggestion
const handleMultipleItemsDeselect = (items: MultiselectItem[], restage = false) => {
```

very nit: can you change the name of the function so it's in same format as `handleMultipleItemsSelect` (`Items` in plural)

File: src/components/KMultiselect/KMultiselect.vue [hunk 2]
```diff
@@ -856,11 +907,23 @@ watch(filteredItems, () => {
...
+    }
+
+    const previouslySelectedItems = new Set<string>(oldVal)
+    const currentlySelectedItems = new Set<string>(newVal)
+
+    const selectedAndPresentItems = unfilteredItems.value.filter((item: MultiselectItem) => currentlySelectedItems.has(item.value))
+    const deselectedItems = selectedItems.value.filter((item: MultiselectItem) => !currentlySelectedItems.has(item.value) && previouslySelectedItems.has(item.value))
+
+    if (deselectedItems.length) {
+      handleMultipleItemDeselect(deselectedItems)
```
Comment (portikM) [inline src/components/KMultiselect/KMultiselect.vue:922]:
```suggestion
      handleMultipleItemsDeselect(deselectedItems)
```

see the comment above

File: src/components/KMultiselect/KMultiselect.vue [hunk 3]
```diff
@@ -461,15 +461,31 @@ const uniqueFilterStr = computed((): boolean => {
...
+// The items visible in the main part of the component.
 const visibleSelectedItemsStaging = ref<MultiselectItem[]>([])
+
+// The items in the "overflow" part of the component.
 const invisibleSelectedItemsStaging = ref<MultiselectItem[]>([])
+
+// A set of the values in the "overflow" part of the component.
 const invisibleSelectedItemsStagingSet = new Set<string>()
+
+// Used to store the results of the determination of which items are visible.
```
Comment (portikM) [inline src/components/KMultiselect/KMultiselect.vue:483]: Thanks for adding comments here!

**Review by portikM — COMMENTED** [review]
Looks good to me, functionally tested in the sandbox. Just one small nit-picking suggestion.

**Review by portikM — APPROVED** [review]
LGTM 🚀

**Pull request merged**